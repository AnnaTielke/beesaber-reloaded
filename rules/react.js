module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
