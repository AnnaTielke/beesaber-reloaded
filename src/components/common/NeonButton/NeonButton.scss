// Utilities
@import '../../../tokens/definitions/scssVariables';

#neon-button {
  display: flex;
  justify-content: center;
  margin: 0 10px;

  body {
    align-items: center;
    background: $bg-color;
    display: flex;
    font-family: 'consolas', sans-serif;
    justify-content: center;
    margin: 0;
    min-height: 100vh;
    padding: 0;
  }

  #button {
    color: $main-blue;
    display: inline-block;
    font-size: 20px;
    letter-spacing: 2px;
    overflow: hidden;
    padding: 5px 30px;
    position: relative;
    text-align: center;
    text-decoration: none;
    text-transform: uppercase;
    transition: 0.2s;
  }

  #button:hover {
    background: $main-blue;
    box-shadow: 0 0 10px $main-blue, 0 0 40px $main-blue, 0 0 80px $main-blue;
    color: $main-blue-light;
    transition-delay: 1s;
  }

  #button span {
    display: block;
    position: absolute;
  }

  #button span:nth-child(1) {
    background: linear-gradient(90deg, transparent, $main-blue);
    height: 2px;
    left: -100%;
    top: 0;
    width: 100%;
  }

  #button span:nth-child(2) {
    background: linear-gradient(270deg, transparent, $main-blue);
    bottom: 0;
    height: 2px;
    right: -100%;
    width: 100%;
  }

  #button span:nth-child(3) {
    background: linear-gradient(180deg, transparent, $main-blue);
    height: 100%;
    right: 0%;
    top: -100%;
    width: 2px;
  }

  #button span:nth-child(4) {
    background: linear-gradient(360deg, transparent, $main-blue);
    bottom: -100%;
    height: 100%;
    left: 0%;
    width: 2px;
  }

  #button:hover span:nth-child(1) {
    left: 100%;
    transition: 1s;
  }

  #button:hover span:nth-child(2) {
    right: 100%;
    transition: 1s;
    transition-delay: 0.5s;
  }

  #button:hover span:nth-child(3) {
    top: 100%;
    transition: 1s;
    transition-delay: 0.25s;
  }

  #button:hover span:nth-child(4) {
    bottom: 100%;
    transition: 1s;
    transition-delay: 0.75s;
  }
}
