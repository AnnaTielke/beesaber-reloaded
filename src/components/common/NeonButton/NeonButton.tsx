import { MDBIcon } from 'mdb-react-ui-kit';
import styled from 'styled-components';

import NeonText from '@/components/common/NeonText/NeonText';

import type { ElementType } from '../NeonText/NeonText';

type Props = {
  as?: ElementType;
  logo: string;
  text: string;
};

const Container = styled.div`
  display: flex;
  justify-content: center;
  margin: 0 10px;

  body {
    align-items: center;
    background: #00111f;
    display: flex;
    font-family: 'consolas', sans-serif;
    justify-content: center;
    margin: 0;
    min-height: 100vh;
    padding: 0;
  }
`;
const Button = styled.div`
  color: rgb(0 0 255);
  display: inline-block;
  font-size: 20px;
  letter-spacing: 2px;
  overflow: hidden;
  padding: 0 30px;
  position: relative;
  text-align: center;
  text-decoration: none;
  text-transform: uppercase;
  transition: 0.2s;

  :hover {
    background: rgb(0 0 255);
    box-shadow: 0 0 10px rgb(0 0 255), 0 0 40px rgb(0 0 255),
      0 0 80px rgb(0 0 255);
    color: #fff0f0;
    transition-delay: 1s;
  }

  span {
    display: block;
    position: absolute;
  }

  span:nth-child(1) {
    background: linear-gradient(90deg, transparent, rgb(0 0 255));
    height: 2px;
    left: -100%;
    top: 0;
    width: 100%;
  }

  span:nth-child(2) {
    background: linear-gradient(270deg, transparent, rgb(0 0 255));
    bottom: 0;
    height: 2px;
    right: -100%;
    width: 100%;
  }

  span:nth-child(3) {
    background: linear-gradient(180deg, transparent, rgb(0 0 255));
    height: 100%;
    right: 0%;
    top: -100%;
    width: 2px;
  }

  span:nth-child(4) {
    background: linear-gradient(360deg, transparent, rgb(0 0 255));
    bottom: -100%;
    height: 100%;
    left: 0%;
    width: 2px;
  }

  :hover span:nth-child(1) {
    left: 100%;
    transition: 1s;
  }

  :hover span:nth-child(2) {
    right: 100%;
    transition: 1s;
    transition-delay: 0.5s;
  }

  :hover span:nth-child(3) {
    top: 100%;
    transition: 1s;
    transition-delay: 0.25s;
  }

  :hover span:nth-child(4) {
    bottom: 100%;
    transition: 1s;
    transition-delay: 0.75s;
  }
`;

const NeonButton = ({ as, logo, text }: Props): JSX.Element => (
  <Container>
    <Button role="button">
      <span></span>
      <span></span>
      <span></span>
      <span></span>
      <MDBIcon fab icon={logo || ''} />{' '}
      <NeonText glow as={as} titleColor={'blue'}>
        {text}
      </NeonText>
    </Button>
  </Container>
);

export default NeonButton;
