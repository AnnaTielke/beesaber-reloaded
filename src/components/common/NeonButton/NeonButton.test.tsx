import { render } from '@testing-library/react';
import React from 'react';

import NeonButton from './NeonButton';

describe('NeonButton', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <NeonButton logo={'google'} text={'Button Text'} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
