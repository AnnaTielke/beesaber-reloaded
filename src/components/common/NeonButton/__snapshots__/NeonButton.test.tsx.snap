// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NeonButton should match the snapshot 1`] = `
<div
  id="neon-button"
>
  <div
    id="button"
  >
    <span />
    <span />
    <span />
    <span />
    <i
      class="fab fa-google"
    />
     
    Button Text
  </div>
</div>
`;
